scripts/benchmarks/benches/syclbench.py

```diff
@@ -0,0 +1,379 @@
+import csv
+import io
+from utils.utils import run, git_clone, create_build_path
+from .base import Benchmark
+from .result import Result
+from .options import options
+
+class SyclBench:
+    def __init__(self, directory):
+        self.directory = directory
+        self.built = False
+        self.setup()
+        return
+
+    def setup(self):
+        build_path = os.path.join(self.directory, 'sycl-bench-build')
+        self.bins = build_path
+
+        if self.built or not options.rebuild:
+            return
+
+        create_build_path(build_path, '')
+
+        repo_path = git_clone(self.directory, "sycl-bench-repo", "https://github.com/mateuszpn/sycl-bench.git", "1e6ab2cfd004a72c5336c26945965017e06eab71")
+        print(f"Clone repo to: {repo_path}")
+
+        print(f"Build path: {build_path}")
+
+        configure_command = [
+            "cmake",
+            f"-B {build_path}",
+            f"-S {repo_path}",
+            f"-DCMAKE_BUILD_TYPE=Release",
+            f"-DCMAKE_CXX_COMPILER={options.sycl}/bin/clang++",
+            f"-DCMAKE_C_COMPILER={options.sycl}/bin/clang",
+            f"-DSYCL_IMPL=dpcpp"
+        ]
+
+        print(f"Run {configure_command}")
+        run(configure_command, add_sycl=True)
+        print(f"Run cmake --build {build_path}")
+        run(f"cmake --build {build_path} -j", add_sycl=True)
+
+        self.built = True
+        return
+    
+class SyclBenchmark(Benchmark):
+    def __init__(self, bench, name, test):
+        self.bench = bench
+        self.bench_name = name
+        self.test = test
+        super().__init__(bench.directory)
+
+    def bin_args(self) -> list[str]:
+        return []
+
+    def extra_env_vars(self) -> dict:
+        return {}
+
+    def unit(self):
+        return "ms"
+
+    def setup(self):
+        self.bench.setup()
+        self.benchmark_bin = os.path.join(self.bench.bins, self.bench_name)
+
+    def run(self, env_vars) -> Result:
+        outputfile = f"{self.bench.directory}/{self.test}.csv"
+        command = [
+            f"{self.benchmark_bin}",
+            f"--warmup-run",
+            f"--num-runs=3",
+            f"--output={outputfile}"
+        ]
+        bin_dir = self.bench.bins
+
+        command += self.bin_args()
+        env_vars.update(self.extra_env_vars())
+
+        print(f"Command: {command}")
+        result = self.run_bench(command, env_vars)
+
+        with open(outputfile, 'r') as f:
+            reader = csv.reader(f)
+            res_list = []
+            for row in reader:
+                if not row[0].startswith('#'):
+                    res_list.append(
+                        Result(label=row[0], 
+                               value=float(row[12]) * 1000, # convert to ms 
+                               command=command, 
+                               env=env_vars, 
+                               stdout=result))
+            
+            median_list = []
+            for label in set(result.label for result in res_list):
```

Comment:
this is already done for all the benchmarks in main.py. why repeat it?

Reply:
Removed